src/StatementBase.php

```diff
@@ -60,7 +60,7 @@ public function __construct() {
 
     private function _asVersion(&$result, $version) {
         foreach ($result as $property => $value) {
-            if ($value !== false && empty($value)) {
+            if ($value === null || $value === array()) {
```

Comment:
Instead, check for the two things we specifically want to treat as unset values - null and an empty array.